src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,142 @@
-package co.inventorsoft.academy.collections.model;
+package main.java.co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
+
+;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Set;
+import java.util.Iterator;
+import java.util.Collection;
+import java.util.function.Function;
 
 public class Range<T> implements Set<T> {
+    private List<T> arrayList = new ArrayList<>();
+
+    public Range() {
+        this.arrayList = arrayList;
+    }
+
 
     public int size() {
-        return 0;
+        return arrayList.size();
     }
 
     public boolean isEmpty() {
-        return false;
+        return arrayList.isEmpty();
     }
 
     public boolean contains(Object o) {
-        return false;
+        return arrayList.contains(o);
     }
 
     public Iterator<T> iterator() {
-        return null;
+        return (Iterator<T>) arrayList.iterator();
     }
 
     public Object[] toArray() {
-        return new Object[0];
+        return arrayList.toArray();
     }
 
     public <T1> T1[] toArray(T1[] a) {
         return null;
     }
 
     public boolean add(T t) {
-        return false;
+        if (!arrayList.contains(t)) {
+            return arrayList.add(t);
+        } else {
+            return false;
+        }
     }
 
     public boolean remove(Object o) {
-        return false;
+        return arrayList.remove(o);
     }
 
     public boolean containsAll(Collection<?> c) {
-        return false;
+        return arrayList.containsAll(c);
     }
 
     public boolean addAll(Collection<? extends T> c) {
-        return false;
+        Object[] objects = c.toArray();
+        boolean containElement = false;
+        for (int i = 0; i < objects.length; i++) {
+            if (!arrayList.contains(objects[i])) {
+                arrayList.add((T) objects[i]);
+                containElement = true;
+            }
+            containElement = false;
+        }
+        return containElement;
     }
 
     public boolean retainAll(Collection<?> c) {
-        return false;
+        return arrayList.retainAll(c);
     }
 
     public boolean removeAll(Collection<?> c) {
-        return false;
+        return arrayList.removeAll(c);
     }
 
     public void clear() {
+        arrayList.clear();
+    }
 
+    public static Range<Integer> of(int i, int i1) {
+        // arrayList.clear();
+        Range<Integer> integers = new Range<>();
+        if (i < i1) {
+            for (int j = i; j <= i1; j++) {
+                integers.add(j);
+            }
+        }
+        return integers;
     }
+
+    public static Range<Float> of(float i, float i1) {
+        Range<Float> floats = new Range<>();
```

Comment:
user overloaded method of with 3 parameters. Just call it. The same on of(double, double)..